src/app/model/ChessEngine.ts

```diff
@@ -67,4 +72,49 @@ export class ChessEngine implements ChessModel {
   private changeTurn(side: Side): Side {
     return side === Side.WHITE ? Side.BLACK : Side.WHITE;
   }
+
+  private willBeKingChecked(player: Player, chessboard: Chessboard, squareFrom: Square, squareTo: Square): boolean {
+    let isCheckedSquareFlag = false;
+
+    const proposedChessboard = _.cloneDeep(chessboard);
+    const { squaresWithPiece: proposedSquaresWithPieces } = proposedChessboard;
```

Comment:
Fajnie byłoby wydzielić jakąś metodę, która mówi, czy Król dla danej strony jest szachowany przy danym ułożeniu szachownicy. Wtedy możemy to wykorzystać też w przypadku zwracania informacji o szachowaniu króla przeciwnika po ruchu przy okazji kolejnych tasków.

Reply:
pomyślę :)